test/org/sagebionetworks/bridge/services/AuthenticationServiceMockTest.java

```diff
@@ -0,0 +1,284 @@
+
+    @Mock
+    private CacheProvider cacheProvider;
+    @Mock
+    private BridgeConfig config;
+    @Mock
+    private ConsentService consentService;
+    @Mock
+    private ParticipantOptionsService optionsService;
+    @Mock
+    private AccountDao accountDao;
+    @Mock
+    private ParticipantService participantService;
+    @Mock
+    private SendMailService sendMailService;
+    @Mock
+    private StudyService studyService;
+    @Mock
+    private PasswordResetValidator passwordResetValidator;
+    @Mock
+    private Study study;
+    @Mock
+    private Account account;
+    @Mock
+    private UserSession session;
+    
+    @Captor
+    private ArgumentCaptor<String> stringCaptor;
+    @Captor
+    private ArgumentCaptor<EmailSignInEmailProvider> providerCaptor;
+    
+    @Spy
+    private AuthenticationService service;
+
+    @Before
+    public void before() {
+        service.setCacheProvider(cacheProvider);
+        service.setBridgeConfig(config);
+        service.setConsentService(consentService);
+        service.setOptionsService(optionsService);
+        service.setAccountDao(accountDao);
+        service.setPasswordResetValidator(passwordResetValidator);
+        service.setParticipantService(participantService);
+        service.setSendMailService(sendMailService);
+        service.setStudyService(studyService);
+
+        doReturn(study).when(studyService).getStudy(STUDY_ID);
+        doReturn(STUDY_ID).when(study).getIdentifier();
+        doReturn(true).when(study).isEmailSignInEnabled();
+        doReturn(new StudyIdentifierImpl(STUDY_ID)).when(study).getStudyIdentifier();
+        doReturn(new EmailTemplate("subject","body",MimeType.TEXT)).when(study).getEmailSignInTemplate();
+        doReturn("sender@sender.com").when(study).getSupportEmail();
+        doReturn("Sender").when(study).getName();
+    }
+    
+    @Test
+    public void requestEmailSignIn() throws Exception {
+        doReturn(account).when(accountDao).getAccountWithEmail(study, RECIPIENT_EMAIL);
+        
+        service.requestEmailSignIn(SIGN_IN_REQUEST);
+        
+        verify(cacheProvider).getString(stringCaptor.capture());
+        assertEquals(CACHE_KEY, stringCaptor.getValue());
+        
+        verify(accountDao).getAccountWithEmail(study, RECIPIENT_EMAIL);
+        
+        verify(cacheProvider).setString(eq(CACHE_KEY), stringCaptor.capture(), eq(60));
+        assertNotNull(stringCaptor.getValue());
+
+        verify(sendMailService).sendEmail(providerCaptor.capture());
+        
+        EmailSignInEmailProvider provider = providerCaptor.getValue();
+        MimeTypeEmail email = provider.getMimeTypeEmail();
+        assertEquals(RECIPIENT_EMAIL, email.getRecipientAddresses().get(0));
+        assertEquals("\"Sender\" <sender@sender.com>", email.getSenderAddress());
+        assertEquals("body", email.getMessageParts().get(0).getContent());
```

Comment:
This test seems to test the implementation of EmailSignInEmailProvider, not AuthenticationService.requestEmailSignIn(). It would be better to verify that the email provider contains the values we passed in (study, email, token). If necessary, we may need to expose some getters for tests.